Idea: Tracks from Native client.

wzieba opened the pull request:
Closes: #17 

@Ecarrion this PR allows clients to integrate sending Tracks event from React classes via native clients Tracks implementations.

In the case of Android, injection of class that satisfies `AnalyticsBridge` interface happens [here](https://github.com/woocommerce/WooCommerce-Shared/compare/issue/17-tracks_from_native_client?expand=1#diff-119b0677854bb809b846ccb122635a04a9aea8ff40a8e44eadb32d952188a043R28) - we route dependencies via `Application` class, with which every `Activity` (including `ReactActivity`) can communicate.

Could you please verify if this solution is feasible for iOS?

wzieba committed:
feat: introduce native bridge and package

wzieba committed:
feat: send Tracks event on shipping zones successful fetch & show

Ecarrion committed:
Update WooCommerceShared to receive an external AnalyticsProvider

Ecarrion committed:
Update demo app to pass the new dependencies

Ecarrion committed:
Improve init methods

Ecarrion commented:
Hey @wzieba, I have added some commits to add support for the analytics native module in iOS following the approach you presented here.

I can confirm that it works on iOS, feel free to try it by running the iOS app on the `feature/rn-integration` branch.

The only thing I found weird was the name of the native module. Currently, it is named `MyReactNativeBridge` and it should probably be named `NativeAnalytics` or just `Analytics`. What do you think?

<img width="758" alt="Screenshot 2023-06-28 at 4 15 02 AM" src="https://github.com/woocommerce/WooCommerce-Shared/assets/562080/bcd5590f-1690-410d-b8da-87b481767e09">

wzieba commented:
Thanks @Ecarrion ! Awesome you made it work 👍 

> The only thing I found weird was the name of the native module.

You're right - it's a leftover from the chat. I'll address it, thanks!

Ecarrion commented:
> You're right - it's a leftover from the chat. I'll address it, thanks!

Cool @wzieba, let me know when you do that so I can change the native module name on iOS!

Ecarrion committed:
Fix incorrect token parameter

wzieba committed:
Merge remote-tracking branch 'origin' into issue/17-tracks_from_native_client

wzieba committed:
feat: fake implementation of the Analytics bridge for Demo Android app

wzieba committed:
style: rename "WooReactNativeBridge" / "MyReactNativeBridge" to "AnalyticsModule"

wzieba commented:
@Ecarrion I renamed the module to "AnalyticsModule". When working on that, occasionally the module was not generated at all and logging `NativeModules` was resulting with empty array. After I've cleaned cache (`make dev -- --reset-cache`) the module was generated 👍

Ecarrion committed:
Update iOS module name

Ecarrion commented:
@wzieba Cool, iOS code updated!
